code/__HELPERS/matrices.dm

```
@@ -193,3 +193,39 @@ round(cos_inv_third+sqrt3_sin, 0.001), round(cos_inv_third-sqrt3_sin, 0.001), ro
 		for(x in 1 to 4)
 			output[offset+x] = round(A[offset+1]*B[x] + A[offset+2]*B[x+4] + A[offset+3]*B[x+8] + A[offset+4]*B[x+12]+(y==5?B[x+16]:0), 0.001)
 	return output
+
+///Converts RGB shorthands into RGBA matrices complete of constants rows (ergo a 20 keys list in byond).
+/proc/color2fullRGBAmatrix(color)
+	if(istext(color))
+		var/list/L = ReadRGB(color)
+		if(!L)
+			return color_matrix_identity() //invalid format
```

Why silently fail instead of something like `CRASH`?